airbyte-integrations/connectors/source-google-ads-asaas/source_google_ads/source.py

```diff
@@ -0,0 +1,231 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+from logging import exception
+from typing import Any, List, Mapping, Tuple, Union
+
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from google.ads.googleads.errors import GoogleAdsException
+from pendulum import timezone
+from pendulum.tz.timezone import Timezone
+import pendulum
+import re
+from dateutil.relativedelta import relativedelta
+
+from .custom_query_stream import CustomQuery
+from .google_ads import GoogleAds
+from .streams import (
+    AccountPerformanceReport,
+    Accounts,
+    AdGroupAdReport,
+    AdGroupAds,
+    AdGroups,
+    Campaigns,
+    CampaignMetricsReport,
+    ClickView,
+    DisplayKeywordPerformanceReport,
+    DisplayTopicsPerformanceReport,
+    GeographicReport,
+    KeywordReport,
+    ShoppingPerformanceReport,
+    UserLocationReport,
+)
+
+
+class PatternHandler(object):
+    """Pattern Handler.
+
+    Class handles the patterns for textual dates
+    """
+
+    def __init__(self, pattern, handler_func) -> None:
+        """Construct Object.
+
+        Args:
+            pattern (str): Regular expression for textual date.
+            handler_func (function): Handler function for transformation.
+        """
+        self.pattern = pattern
+        self.handler_func = handler_func
+
+    def parse(self, date_text):
+        """Parse textual date.
+
+        Args:
+            date_text (str): Textual date.
+
+        Returns:
+            datetime: Date parsed.
+        """
+        date_match = re.fullmatch(self.pattern, date_text)
+
+        if date_match:
+            today = pendulum.today(tz="UTC")
+            expr = self.handler_func(date_match.group(1))
+            return today + relativedelta(**expr)
+
+
+class DateHandler(object):
+    """Date Handler.
+
+    Class handles the dates
+    """
+
+    def __init__(self, patterns: list) -> None:
```

Comment:
```suggestion
    def __init__(self, patterns: List[PatternHandler]) -> None:
```
Melhorar o typing do `patterns`.